src/wallet/rpcwallet.cpp

```diff
@@ -1598,6 +1601,47 @@ void ListTransactions(CWallet* const pwallet, const CWalletTx& wtx, const std::s
             }
         }
     }
+
+    /** RVN START */
+    if (listAssetsReceived.size() > 0 && wtx.GetDepthInMainChain() >= nMinDepth)
+    {
+        for (const CAssetOutputEntry& data : listAssetsReceived) {
+            UniValue entry(UniValue::VOBJ);
+
+            entry.push_back(Pair("asset_type", data.type));
+            entry.push_back(Pair("asset_name", data.assetName));
+            entry.push_back(Pair("amount", ValueFromAmount(data.amount)));
+            entry.push_back(Pair("destination", EncodeDestination(data.destination)));
+            entry.push_back(Pair("vout", data.vout));
+            entry.push_back(Pair("category", "received"));
+            retAssets.push_back(entry);
+        }
+    }
+
+
+    if ((!listAssetsSent.empty() || nFee != 0) && (fAllAccounts || strAccount == strSentAccount))
+    {
+        for (const CAssetOutputEntry& data : listAssetsSent) {
+            UniValue entry(UniValue::VOBJ);
+
+            entry.push_back(Pair("asset_type", data.type));
+            entry.push_back(Pair("asset_name", data.assetName));
+            entry.push_back(Pair("amount", ValueFromAmount(data.amount)));
+            entry.push_back(Pair("destination", EncodeDestination(data.destination)));
+            entry.push_back(Pair("vout", data.vout));
+            entry.push_back(Pair("category", "send"));
```

Comment:
I'd use `sent` instead because it compliments `received`.  🔤🤓

Reply:
I was using the same words that the current rpc command displays. Any thing we can do to make integration with current block explorers easy would be the best. I can change it if it would make more sense to the user though.

Reply:
no that makes sense